instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/BodyHandlerWrapper.java

```diff
@@ -30,7 +30,7 @@ public BodyHandlerWrapper(BodyHandler<T> delegate, Context context) {
     if (subscriber instanceof BodySubscriberWrapper) {
       return subscriber;
     }
-    return new BodySubscriberWrapper<>(delegate.apply(responseInfo), context);
+    return new BodySubscriberWrapper<>(subscriber, context);
```

Comment:
Can you please explain, why the previous version caused the memory leak?

Reply:
On the line before the diff there is `BodySubscriber<T> subscriber = delegate.apply(responseInfo);` Apparently in case of ktor the result of `delegate.apply` which is a JavaHttpResponseBodySubscriber https://github.com/ktorio/ktor/blob/main/ktor-client/ktor-client-java/jvm/src/io/ktor/client/engine/java/JavaHttpResponseBodyHandler.kt can't just be thrown away without closing it. Anyway I believe that I'm the author of this bug, it should have looked like this from the start.